docs/bootstrap.md

```diff
@@ -0,0 +1,83 @@
+## Bootstrap
+
+`Bootstrapping` refers to a self-starting process that is supposed to proceed without external input.
+Mainflux platform supports bootstrapping process, but some of the preconditions need to be fulfilled in advance.
+
```

Comment:
Add architecture diagram and show where Bootstrap Service sits in the system.

Reply:
Something similar to this: https://medium.com/@vrmvrm/device-key-distribution-with-lightweight-m2m-36cdc12e5711

Reply:
Also - exmplain difference between Bootstrap and ordinary provisioning.